content/docs/reactivesearch/v3/advanced/stateprovider.md

```diff
@@ -113,6 +113,24 @@ Here is a list of all the valid keys:
 -   **error** returns the error
 -   **hits** An array of hits obtained from the applied query.
 -   **aggregations** An array of aggregations obtained from the applied query.
+-   **promotedData** An array of promoted results obtained from the applied query. [Read More](/docs/search/Rules#part-1-introduction).
+-   **resultStats**: `object`
+    An object with the following properties which can be helpful to render custom stats:
+    -   **`numberOfResults`**: `number`
+        Total number of results found
+    -   **`numberOfPages`**: `number`
+        Total number of pages found based on current page size
+    -   **`currentPage`**: `number`
+        Current page number for which data is being rendered
+    -   **`time`**: `number`
+        Time taken to find total results (in ms)
+    -   **`displayedResults`**: `number`
+        Number of results displayed in current view
+    -   **`hidden`**: `number`
+        Total number of hidden results found
+    -   **`promoted`**: `number`
+        Total number of promoted results found
+        > Note <br /> `numberOfPages`, `currentPage` and `displayedResults` are only available for `ReactiveList` component.
```

Comment:
@ShahAnuj2610 Are you sure that we provide `currentPage` in StateProvider for RL. If yes then can you ref. the code.

Reply:
fixed